path/path_test.go

```diff
@@ -28,3 +28,27 @@ func TestPathParsing(t *testing.T) {
 		}
 	}
 }
+
+func TestPathCleaning(t *testing.T) {
+	cases := map[string]string{
+		// .. should strip the preceding path segment
```

Comment:
agreed @willglynn -- and agreed with plan9 folks: 

> The parent of a directory `X`, `X/..`, is the same directory that would obtain if we instead accessed the directory named by stripping away the last path name element of `X`. 

So this basically means that `..` is relative to the access path, correct? (this makes perfect sense to me)